localstack/services/dynamodb/provider.py

```diff
@@ -549,60 +551,67 @@ def describe_table(self, context: RequestContext, table_name: TableName) -> Desc
-                )
+            if replica_updates := update_table_input.get("ReplicaUpdates"):
+                store = get_store(context.account_id, global_table_region)
+
+                # Dict with source region to set of replicated regions
+                replicas: dict[str, set(str)] = store.REPLICA_UPDATES.get(table_name, {})
+                replicas.setdefault(global_table_region, set())
+
+                for replica_update in replica_updates:
+                    for key, details in replica_update.items():
+                        # Replicated region
+                        target_region = details.get("RegionName")
+
+                        # Check if replicated region is valid
+                        if target_region not in get_valid_regions_for_service("dynamodb"):
+                            raise ValidationException(f"Region {target_region} is not supported")
 
-                # get dict of replicas for the table => {original_region: {regions}}
-                original_region = context.region
-                table_replicas[original_region] = region_replicas = (
-                    table_replicas.get(original_region) or set()
-                )
-                for repl_update in update_table_input["ReplicaUpdates"]:
-                    for key, details in repl_update.items():
-                        # target region
-                        region = details.get("RegionName")
                         match key:
                             case "Create":
-                                table_replicas[original_region].add(region)
-                            case "Update":
-                                table_replicas[original_region] = set(
-                                    [r for r in region_replicas if r == region]
-                                )
+                                if target_region in replicas[global_table_region]:
+                                    raise ValidationException(
+                                        f"Failed to create a the new replica of table with name: '{table_name}' because one or more replicas already existed as tables."
+                                    )
+                                replicas[global_table_region].add(target_region)
                             case "Delete":
-                                table_replicas[original_region] = set(
-                                    [r for r in region_replicas if r != region]
-                                )
+                                try:
+                                    replicas[global_table_region].remove(target_region)
+                                    if len(replicas[global_table_region]) == 0:
+                                        # Removing the set indicates that replication is disabled
+                                        replicas.pop(global_table_region)
+                                except KeyError:
+                                    raise ValidationException(
```

Comment:
nit: do we have a test that triggers this validation exception?

Reply:
Added in 679a1c0e :heavy_check_mark: